traffic-sniffer/lib/request-metadata/setup-file.ts

```diff
@@ -0,0 +1,105 @@
+
+export type SnifferConfigSetup = SnifferConfig & { isStarted: boolean }
+
+export interface ConfigLoader {
+    configData: SnifferConfigSetup[];
+
+    getSetup(): SnifferConfigSetup[];
+    update(existingId: string, newConfig: SnifferConfig, isStarted: boolean): void;
+    addSniffer(snifferConfig: SnifferConfig): void;
+    removeSniffer(port: number): void;
+    setIsStarted(snifferId: string, isStarted: boolean):  void;
+}
+
+export class FileConfig implements ConfigLoader {
+    configData: SnifferConfigSetup[];
+
+    constructor() {
+        this.validateSetupFileExists()
+        this.configData = this.readSetupFileData();
+    }
+
+    getSetup() {
+        return this.configData;
+    }
+
+    update(existingId: string, newConfig: SnifferConfig, isStarted: boolean) {
+        const foundIndex = this.configData.findIndex((item) => item.id === existingId);   
+        if (foundIndex === -1) {
+          throw new Error("item was not found")
+        }
+        this.configData[foundIndex] = this.createSnifferSetup(newConfig, isStarted);
+        this.writeToSetupFile()
+      }
+    
+    validateSetupFileExists() {
+        if (!fs.existsSync(setupFilePath)) {
+            fs.writeFileSync(setupFilePath, JSON.stringify([]), { flag: 'w'});
+            }
+    }  
+    
+
+    readSetupFileData(): SnifferConfigSetup[] {
+        let setupData: SnifferConfigSetup[] = []
+        try {
+            setupData = JSON.parse(fs.readFileSync(setupFilePath, "utf-8"));
+        }
+        catch (err) {
+            throw new Error("setup file is not in right format!")
+        }
+        return setupData
+    }
+
+    addSniffer(snifferConfig: SnifferConfig) {
+        const addedObj = this.createSnifferSetup(snifferConfig, false)
+    
+        if (this.configData.findIndex((item) => item.id === snifferConfig.id ) === -1) {
+            this.configData.push(addedObj)
+            this.writeToSetupFile()
+        }
+        else {
+            console.log("sniffer already listed");
+        }
+    }
+
+    removeSniffer(port: number) {
+        const foundIndex = this.configData.findIndex((item) => item.port === port);   
+        if (foundIndex === -1) {
+            throw new Error("item was not found")
+        }
+        this.configData.splice(foundIndex, 1);
+        this.writeToSetupFile()
+
+    }
+
+    setIsStarted(snifferId: string, isStarted: boolean) {
+        const foundIndex = this.configData.findIndex((item) => item.id === snifferId);   
+        if (foundIndex === -1) {
+          throw new Error("item was not found")
+        }
+        const updatedSetup = this.configData[foundIndex];
+        updatedSetup.isStarted = isStarted
+        this.configData[foundIndex] = updatedSetup;
+        this.writeToSetupFile()
+    }
+
+    writeToSetupFile() {
+        fs.writeFile(setupFilePath, JSON.stringify(this.configData), (err) => {
+            if(err) { throw new Error("unable to update setup file"); }
+        })
```

Comment:
Should we change this is to async/await design?

Reply:
Should we wait for success? 
The last time I talked about it with Ido I changed it to async.

Reply:
You can use 
import fs from 'fs/promises'
And then you can await the writeFIle command.